File: 백준/Silver/2667. 단지번호붙이기/단지번호붙이기.py
import sys

#input = sys.stdin.readline
N = int(input())
graph = []
for i in range(N):
    graph.append(list(map(int, input())))

result = 0
num = 0
answer = []

def dfs(row, column):

    global num

    if row < 0 or column < 0 or row > N-1 or column > N-1:
        return False

    if graph[row][column] == 1:
        graph[row][column] = 0
        dfs(row - 1, column)
        dfs(row + 1, column)
        dfs(row, column + 1)
        dfs(row, column - 1)

        num += 1

        return True

    return False

for row in range(N):
    for column in range(N):
        if dfs(row, column) == True:
            result += 1
            answer.append(num)
            num = 0

answer.sort()
print(result)
for a in answer:
    print(a)

File: 백준/Silver/2667. 단지번호붙이기/단지번호붙이기.js
const { dir } = require("console");
const fs = require("fs");
const stdin = (
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString()
    : fs.readFileSync("../input.txt").toString()
).split("\n");

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const N = Number(input().trim());

//const area = Array.from(Array(N), () => Array(N).fill(false));
const visited = Array.from(Array(N), () => Array(N).fill(false));

const direction = [
  [0, 1],
  [0, -1],
  [1, 0],
  [-1, 0],
];

const area = [];

for (let i = 0; i < N; i++) {
  area.push(
    input()
      .trim()
      .split("")
      .map((v) => parseInt(v))
  );
}

// y + d[0] >= 0 && y + d[0] < N && x + d[1] >= 0 && x + d[1] < M;

function dfs(y, x, originY, originX) {
  for (const d of direction) {
    if (
      y + d[0] >= 0 &&
      y + d[0] < N &&
      x + d[1] >= 0 &&
      x + d[1] < N &&
      !visited[y + d[0]][x + d[1]] &&
      area[y + d[0]][x + d[1]]
		) {
			visited[y + d[0]][x + d[1]] = true;
			area[originY][originX] += 1;
			dfs(y + d[0], x + d[1], originY, originX);
    }
  }
}

const answer = [];

for (let y = 0; y < N; y++) {
  for (let x = 0; x < N; x++) {
    if (!visited[y][x] && area[y][x]) {
      visited[y][x] = true;
			dfs(y, x, y, x);
			answer.push(area[y][x]);
    }
  }
}

answer.sort((x, y) => x - y);
console.log(answer.length);
for (const a of answer) {
	console.log(a);
}